File: frontend/src/types/readiness.ts
export interface ReadinessOverview {
  target_role: { id: string; name: string; slug: string } | null
  score: number | null
  has_minimum_evidence: boolean
  evidence_strength: string
  core_coverage: { covered: number; total: number }
  skills: SkillReadinessItem[]
  strong_skills: string[]
  developing_skills: string[]
  missing_skills: string[]
  why_breakdown: WhyBreakdownItem[]
  trend: { score: number; created_at: string; evidence_strength: string }[]
  recommended_actions: RecommendationAction[]
  message: string | null
}

export interface SkillReadinessItem {
  skill_id?: string
  skill_name?: string
  skill_slug?: string
  importance?: string
  readiness: number
  effective_score: number
  evidence_strength: string
  status: string
  sources: { source: string; score: number; activity_count: number }[]
}

export interface WhyBreakdownItem {
  skill: string
  importance: string
  weight_percent: number
  readiness: number
  effective_score: number
  evidence_strength: string
}

export interface RecommendationAction {
  title: string
  description: string
  reason: string
  skill: string | null
  priority: string
  href: string
  action_type: string
}

export interface MistakeItem {
  id: string
  source_type: string
  source_id: string
  title: string
  summary: string | null
  mistake_type: string
  occurrence_count: number
  status: string
  first_seen_at: string
  last_seen_at: string
  retry_href: string | null
  context: Record<string, unknown> | null
}

export interface MistakeSummary {
  open_count: number
  repeated_count: number
  resolved_count: number
  top_weak_topics: { title: string; count: number }[]
}

export interface JobMatch {
  coverage: number | null
  has_sufficient_mapping: boolean
  has_user_evidence?: boolean
  message: string | null
  required: { skill: string; readiness: number; status: string }[]
  preferred: { skill: string; readiness: number; status: string }[]
  why: { factor: string; weight_percent: number; score: number }[]
  evidence_strength?: string
}

File: backend/app/schemas/readiness.py
"""Readiness API schemas."""

from pydantic import BaseModel, Field


class RoleBrief(BaseModel):
    id: str
    name: str
    slug: str


class CoreCoverage(BaseModel):
    covered: int
    total: int


class SkillSourceBreakdown(BaseModel):
    source: str
    score: float
    activity_count: int


class SkillReadinessItem(BaseModel):
    skill_id: str | None = None
    skill_name: str | None = None
    skill_slug: str | None = None
    importance: str | None = None
    weight: float | None = None
    readiness: float = 0
    effective_score: float = 0
    evidence_strength: str = "low"
    status: str = "no_evidence"
    sources: list[SkillSourceBreakdown] = Field(default_factory=list)


class WhyBreakdownItem(BaseModel):
    skill: str
    importance: str
    weight_percent: float
    readiness: float
    effective_score: float
    evidence_strength: str


class TrendPoint(BaseModel):
    score: float
    created_at: str
    evidence_strength: str


class RecommendationAction(BaseModel):
    title: str
    description: str
    reason: str
    skill: str | None = None
    priority: str
    href: str
    action_type: str


class ReadinessOverview(BaseModel):
    target_role: RoleBrief | None = None
    score: float | None = None
    has_minimum_evidence: bool = False
    evidence_strength: str = "low"
    core_coverage: CoreCoverage = Field(default_factory=lambda: CoreCoverage(covered=0, total=0))
    skills: list[SkillReadinessItem] = Field(default_factory=list)
    strong_skills: list[str] = Field(default_factory=list)
    developing_skills: list[str] = Field(default_factory=list)
    missing_skills: list[str] = Field(default_factory=list)
    why_breakdown: list[WhyBreakdownItem] = Field(default_factory=list)
    trend: list[TrendPoint] = Field(default_factory=list)
    recommended_actions: list[RecommendationAction] = Field(default_factory=list)
    message: str | None = None


class SkillProfileItem(BaseModel):
    skill_id: str
    skill: str
    skill_slug: str
    score: float
    effective_score: float
    evidence_strength: str
    activity_count: int
    last_activity_at: str | None = None
    status: str
    sources: list[SkillSourceBreakdown] = Field(default_factory=list)


class RoleComparisonItem(BaseModel):
    role_id: str
    role_name: str
    role_slug: str
    score: float | None = None
    has_minimum_evidence: bool = False
